### PR TITLE
DDP-5497 testboston: add hospitalized questions to longitudinal v2

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

Implements DDP-5497. Three new questions were added for "hospitalization" details. Since these are required questions, we create a new version of longitudinal activity so that only new users/instances will have these required questions (otherwise, we need to think about how to backfill).

Only thing missing is support for hospital name autocompletion, which will be addressed in a separate ticket.

For deployed environments, we will run the new custom task:

```
$ java -Dconfig.file=output-config/application.conf -jar target/StudyBuilder.jar \
  --vars output-config/testboston-vars.conf --substitutions studies/testboston/subs.conf \
  testboston --run-task TestBostonLongitudinalV2
```

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.\
    - Need to run study-builder task.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document
    - Need to run study-builder task.

### CODE REVIEW

File: study-builder/studies/testboston/longitudinal-covid-v2.conf [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+            {
+              "question": ${_includes.longitudinal.hospital_days},
+              "blockType": "QUESTION",
+              "shownExpr": """
+                user.studies["testboston"].forms["LONGITUDINAL_COVID"].instances[specific].questions["LONGITUDINAL_COVID_BEEN_TESTED"].answers.hasOption("YES")
+                && user.studies["testboston"].forms["LONGITUDINAL_COVID"].instances[specific].questions["LONGITUDINAL_COVID_TEST_KIND"].answers.hasOption("NOT_SURE")
+                && user.studies["testboston"].forms["LONGITUDINAL_COVID"].instances[specific].questions["LONGITUDINAL_COVID_TEST_RESULT"].answers.hasOption("POS")
+                && user.studies["testboston"].forms["LONGITUDINAL_COVID"].instances[specific].questions["LONGITUDINAL_HOSPITALIZED"].answers.hasTrue()
+              """
+            },
```
Comment (yufengwng) [inline study-builder/studies/testboston/longitudinal-covid-v2.conf:143]: This set of lines is the new stuff. Rest of file is a copy of `longitudinal-covid.conf` file. I created a new v2 conf file to make it easier to work with.

Comment (ssettipalli) [reply study-builder/studies/testboston/longitudinal-covid-v2.conf:143]: In the new code showExpr: we explicitly checking for condition :
["LONGITUDINAL_COVID_TEST_KIND"].answers.hasOption("NOT_SURE")
We don't want to ask Hospitalized questions if TEST_KIND is "VIRAL_TEST or BLOOD_TEST" ?

Comment (yufengwng) [reply study-builder/studies/testboston/longitudinal-covid-v2.conf:143]: As far as I know, it only shows up for `NOT_SURE`. This is what the attached document in the ticket shows as well.

**Review by ssettipalli — APPROVED** [review]
looks good to me

File: study-builder/studies/testboston/longitudinal-covid-v2.conf (hunk 1 above)
Comment (ssettipalli) [reply study-builder/studies/testboston/longitudinal-covid-v2.conf:143]: ok

**Pull request merged**